Java/jni/native_glue.cc

```diff
@@ -68,12 +193,15 @@ namespace litecore {
             assert(env != nullptr);
             if (js != nullptr) {
                 jboolean isCopy;
-                const char *cstr = env->GetStringUTFChars(js, &isCopy);
+                _jstr = js;
+                _env = env;
+
+                // Can we assume that a copy is always made here for Android?  It seems like the case.
+                char *cstr = (char *)env->GetStringUTFChars(js, &isCopy);
```

Comment:
This is pretty dangerous if the string _isn't_ copied! Add an assertion of some kind that `isCopy` is true.